### PR TITLE
Add scaling and integration test

### PR DESCRIPTION
**Pull request by kelkawi-a**
This PR slightly refactors the logic in the relation hooks with the Temporal server to enable the web UI charm to scale.

Ingress `backend-protocol` has also been reverted to HTTP to terminate the TLS at the ingress level.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -114,15 +126,38 @@ def _on_config_changed(self, event):
...
+
+        Args:
+            event:The event triggered by the restart action
+        """
+        container = self.unit.get_container(self.name)
+        if not container.can_connect():
+            event.defer()
+            return
+
+        self.unit.status = MaintenanceStatus("restarting ui")
```
Comment (AmberCharitos) [inline src/charm.py:141]: Will this just remain in MaintenanceStatus after restart? You could set an active status after restart, or set up some [checks](https://juju.is/docs/sdk/interact-with-pebble#heading--fetch-check-status) to validate the app has restarted

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -28,16 +28,17 @@
...
 async def deploy(ops_test: OpsTest):
     """The app is up and running."""
-    charm = await ops_test.build_charm(".")
-    resources = {"temporal-ui-image": METADATA["containers"]["temporal-ui"]["upstream-source"]}
-
     # Deploy temporal server, temporal admin and postgresql charms.
-    await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME)
     await ops_test.model.deploy(APP_NAME_SERVER, channel="stable")
     await ops_test.model.deploy(APP_NAME_ADMIN, channel="stable")
     await ops_test.model.deploy("postgresql-k8s", channel="edge", trust=True)
```
Comment (AmberCharitos) [inline tests/integration/test_charm.py:34]: Out of curiosity, what does edge version of postgresql-k8s charm allow for that the stable does not?

**Review by AmberCharitos — APPROVED** [review]
LGTM, 1 suggestion and a question for my curiosity.

File: src/charm.py (hunk 1 above)
Comment (mertalpt) [reply src/charm.py:141]: I am also interested in this.

**Review by mertalpt — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (kelkawi-a) [reply src/charm.py:143]: You're right, i've updated a check to set the active status after restarting. Thanks!

File: tests/integration/test_charm.py (hunk 2 above)
Comment (kelkawi-a) [reply tests/integration/test_charm.py]: This was a remnant of the older integration tests when we were using the edge channel. I've updated the integration test to use the 14/stable channel.

**Pull request merged**